slip-0039.md

```diff
@@ -403,6 +408,12 @@ Python wallets with SLIP39 support:
 
     Some individuals have expressed a concern that the inability to convert SLIP-0039 shares to BIP-0039 may lead to vendor lock-in due to slow adoption of SLIP-0039 by hardware wallet vendors. This concern is unwarranted, since even if the conversion to BIP-0039 were possible and a user needed to recover their seed onto a device which does not support SLIP-0039, then they would need to use some conversion tool running on their computer. In that case they might as well simply recover their SLIP-0039 shares in a software wallet running on their computer and send all of their funds to a new seed on their new device. Thus the ability to convert shares to a BIP-0039 mnemonic makes no difference in this respect.
 
+10. <a name="ExtendableBackup"></a>**Extendable backup flag**
+
+    When this flag is set, it indicates that the *id* is not used as salt in the encryption of the master secret. This makes it possible to create multiple sets of shares, such that each set of shares uses a different *id* and each set of shares leads to the same master secret for every passphrase. This is a desirable property, because it allows users to start using their wallet by creating a single-share (1-of-1) scheme and later upgrade to a multi-share scheme while maintaining the same *EMS* and passphrases. In case the user upgrades two or more times to a multi-share scheme, then each set of shares will in all likelihood have a distinct *id*. Thus it will be possible to tell the sets apart and avoid their accidental mixing, which could potentially lead to loss of access to funds, since shares from different sets are incompatible even if they use the same scheme.
+
+    In newly created shares *ext* SHOULD be set to 1. However, since some users already have shares that use *ext* = 0, implementations MUST support master secret decryption using both *ext* values. If *ext* = 0, then implementations MUST NOT allow the intentional creation of another set of shares with the same *id* and *EMS*.
```

Comment:
> implementations MUST NOT allow the intentional creation 

if i were to nitpick, i'd say "SHOULD NOT" here, and even tighten the conditions to "same EMS + same distinguishing metadata" -- explicitly allowing one of the alternative behaviors, such as (a) extending a 1-of-1 backup or (b) bumping the group id for every newly created group

Reply:
I suppose we could go with "SHOULD NOT". According to RFC 2119:
> there may exist valid reasons in particular circumstances when the
   particular behavior is acceptable or even useful, but the full
   implications should be understood and the case carefully weighed
   before implementing any behavior described with this label.

- If a wallet takes adequate precautions, like allowing the upgrade only once in case (a) or guaranteeing strict monotonicity in case (b), while also guaranteeing in both cases that the upgrades cannot be done for recovered seeds, but only for seeds generated in the wallet, then it seems OK to do, see next point. Furthermore we'd need to clarify what exactly "same distinguishing metadata" means, which might be tricky. I wouldn't want to provide this sort of detailed guidance in the spec, so I'd stick with "EMS and id".
- However, what concerns me with being lenient is that if a user presents shares from mixed backups of the same seed, then wallets that are not aware of this possibility may have trouble guiding the user correctly in the recovery process. For example, in case (a) the user starts by entering a share from the newer multi-share backup and then enters the original 1-of-1 share. The wallet not expecting such a scenario will probably raise an exception, because the thresholds do not match, giving a cryptic error message to the user or possibly aborting the whole process, even though the situation is clearly resolvable. A similar problem may occur in case (b) with the group count. I wouldn't want to complicate this specification by outlining such possibilities and requiring wallets to take something like that into account.

Reply:
> As a general note, this is not 100% backwards compatible because the ext flag is not described as being part of the exponent in the earlier text. Perhaps a mention somewhere in the "Design rationale" section for it?

Can you be more specific? You mean saying "note that the ext flag was added in a later revision, previously it was a bit of the iteration exponent", or also talk about the reasons why it was decided to revise SLIP 39 instead of creating a new one, or something else?

Reply:
> You mean saying "note that the ext flag was added in a later revision, previously it was a bit of the iteration exponent",

Just this, yes, in order to point out that you could encounter shares with ext flag set and `"shamir"` customization string, and there is a valid interpretation of that.

Reply:
so basically the question is which is the right thing to do here, informally speaking, "conformant wallets (SHOULD NOT/MUST NOT) produce data with high likelihood of confusing other conformant wallets", right?

I'd still lean towards "SHOULD NOT".
(maybe even drop the "and EMS" part?)
We could call out this problem by saying something like:
> implementations SHOULD NOT allow the intentional creation of another set of shares with the same *id* and *EMS*. Doing so would run the risk of the users trying to recover shares from incompatible groups, which is something that implementations are not required to handle.

Reply:
[af7a2b0](https://github.com/satoshilabs/slips/pull/1724/commits/af7a2b03fe1eca57197c1d0c013e7177be06b522) should address all of the points discussed.